3-mcstas/mcstas-powder-diffraction.ipynb

```diff
@@ -308,83 +574,288 @@
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "4bdb713d-7c0b-471f-b3d0-f4e79aedc549",
-   "metadata": {},
+   "id": "d511289d-4036-465e-8595-b43d2edff0b5",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
    "source": [
-    "Add the chopper with the following parameters:\n",
-    "- `yheight`: 0.05 m\n",
-    "- `radius`: 0.35 m\n",
-    "- `nslit`: 1.0\n",
-    "- `theta_0`: 7.0 deg\n",
-    "- `delay`: To the variable calculated in the instrument (use quotation marks)\n",
-    "- `nu`: A calculation using the `frequency_multiplier` variable, \"frequency_multiplier*14.0\"\n"
+    "### Which component\n",
+    "\n",
+    "Now we need to select what type of component to add to the instrument, here we will need the `DiskChopper` component.\n",
+    "Use the `component_help` method on the instrument object to learn more about this component."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "3c0530ec-33d1-47f6-bc22-a7583dae8148",
-   "metadata": {},
+   "id": "bb7c25b9-8f1e-4090-94ef-824883a78e96",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": [
+     "solution",
+     "hide-cell"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "instrument.available_components(\"optics\")\n",
+    "instrument.component_help(\"DiskChopper\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9699c192-5b0b-42bf-88fd-ad9174afcb9e",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "### Chopper calculations\n",
+    "When adding a chopper one need to perform calculations to obtain the appropriate rotation frequency and phase.\n",
+    "For this exercise, those calculations can be added to the instrument using a function included in the given Python file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6558bebf-9bf4-47d5-aedf-4c046653aa98",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
    "outputs": [],
    "source": [
-    "# simple answer, to be used to skip this question (also sets position, next question)\n",
+    "make_powder_instrument.add_chopper_calculations(instrument)\n",
+    "\n",
+    "# If low on time, its possible to add the calculations and choppper component, solving question 7 and 8. Only run one of the two functions.\n",
     "#make_powder_instrument.add_chopper(instrument)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "8da3828c-da60-4d71-9ec7-e46958d4c177",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "To see what variables are used in the instrument, one can use the `show_variables` method like below. This will included the ones added by the above function."
```

Comment:
```suggestion
    "To see what variables are used in the instrument, one can use the `show_variables` method like below. This will include the ones added by the above function."
```